evalml/pipelines/components/transformers/preprocessing/time_series.py

```diff
@@ -56,16 +62,31 @@ def transform(self, X, y=None):
         Returns:
             pd.DataFrame: Transformed X.
         """
+        if X is None:
+            X = pd.DataFrame()
         # Normalize the data into pandas objects
-        if not isinstance(X, pd.DataFrame):
-            X = pd.DataFrame(X)
-        if y is not None and not isinstance(y, pd.Series):
-            y = pd.Series(y)
-
-        if self.delay_features and not X.empty:
-            X = X.assign(**{f"{col}_delay_{t}": X[col].shift(t)
-                            for t in range(1, self.max_delay + 1)
-                            for col in X})
+        X = _convert_to_woodwork_structure(X)
+        if y is not None:
+            y = _convert_to_woodwork_structure(y)
+
+        if y.logical_type == logical_types.Categorical:
```

Comment:
Wouldn't this check fail if `y` is None?

Reply:
Fixed! 🙏